setup_yarn.sh

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env bash
+
+set -e
+
+require_var() {
+  if [ -z "$1" ]; then
+    >&2 echo "$2"
+    return 1
+  fi
+  return 0
+}
+
+require_env() {
+  require_var "$2" "Missing env var '$1'"
+}
+
+# Linux base64 wraps while macOS does not
+encode() {
+  local os
+  os=$(uname | tr '[:upper:]' '[:lower:]')
+  if [ "$os" == "linux" ]; then
+    echo -n "$1" | base64 --wrap=0
+  else
+    echo -n "$1" | base64
+  fi
+}
+
+setup_yarn() {
+  if [ "$ARTIFACTORY_SETUP_NPM" == "false" ]; then
+    echo "Skipping npm setup because ARTIFACTORY_SETUP_NPM=$ARTIFACTORY_SETUP_NPM"
```

Comment:
```suggestion
  if [ "$ARTIFACTORY_SETUP_YARN" == "false" ]; then
    echo "Skipping yarn setup because ARTIFACTORY_SETUP_YARN=$ARTIFACTORY_SETUP_YARN"
```
Should be toggled by its own variable.